src/sage/rings/polynomial/multi_polynomial.pyx

```diff
@@ -2613,6 +2615,234 @@ cdef class MPolynomial(CommutativeRingElement):
+        is Lorentzian if:
+
+        - `f` is homogeneous;
+
+        - the support of `f` is `M`-convex
+
+        - `f` has degree less than `2`, or if its degree is at least two,
+          the collection of sequential partial derivatives of `f` which are
+          quadratic forms have Gram matrices with at most one positive eigenvalue.
+
+        Note in particular that the zero polynomial is Lorentzian.  Examples of
+        Lorentzian polynomials include homogeneous stable polynomials, volume
+        polynomials of convex bodies and projective varieties, and Schur polynomials
+        after renormalizing the coefficient of each monomial `x^\alpha` by `1/\alpha!`.
+
+        EXAMPLES:
+
+        Renormalized Schur polynomials are Lorentzian, but not in general if the
+        renormalization is skipped::
+
+            sage: P.<x,y> = QQ[]
+            sage: p = (x^2 / 2) + x*y + (y^2 / 2)
+            sage: p.is_lorentzian()
+            True
+            sage: p = x^2 + x*y + y^2
+            sage: p.is_lorentzian()
+            False
+
+        Homogeneous linear forms and constant polynomials with positive
+        coefficients are Lorentzian, as well as the zero polynomial::
+
+            sage: p = x + 2*y
+            sage: p.is_lorentzian()
+            True
+            sage: p = P(5)
+            sage: p.is_lorentzian()
+            True
+            sage: P.zero().is_lorentzian()
+            True
+
+        Inhomogeneous polynomials and polynomials with negative coefficients
+        are not Lorentzian::
+
+            sage: p = x^2 + 2*x + y^2
+            sage: p.is_lorentzian()
+            False
+            sage: p = 2*x^2 - y^2
+            sage: p.is_lorentzian()
+            False
+
+        It is an error to check if a polynomial is Lorentzian if its base ring
+        is not a subring of the real numbers, as the notion is not defined in
+        this case::
+
+            sage: Q.<z,w> = CC[]
+            sage: q = z^2 + w^2
+            sage: q.is_lorentzian()
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: is_lorentzian only implemented for real polynomials
+
+        The method can give a reason for a polynomial failing to be Lorentzian::
+
+            sage: p = x^2 + 2*x + y^2
+            sage: p.is_lorentzian(explain=True)
+            (False, 'inhomogeneous')
+
+        REFERENCES:
+
+        For full definitions and related discussion, see [BrHu2019]_ and
+        [HMMS2019]_.  The second reference gives the characterization of
+        Lorentzian polynomials applied in this implementation explicitly.
+        """
+        # function to handle return value when reason requested
+        def result(val, explanation=None):
+            if explain:
+                return (val, explanation)
+            else:
+                return val
+
+        if not self.base_ring().is_subring(RealField()):
```

Comment:
This will only check if it is contained within RR, with a fixed 53 bits. So this will fail for smaller number of bits:
```
sage: RealField(103).is_subring(RealField())
True
sage: RealField(22).is_subring(RealField())
False
```
Although I think the above is a bug (and should be the other way around as it is relying on a coercion map). I am not sure how useful this test is even to have.